src/irmin/info.ml

```diff
@@ -16,26 +16,35 @@
 
 include Info_intf
 
-module Default = struct
+module Make (V : Version.S) = struct
   type author = string [@@deriving irmin]
   type message = string [@@deriving irmin]
-
-  type t = { date : int64; author : author; message : message }
+  type version = V.t [@@deriving irmin]
+
+  type t = {
+    date : int64;
+    author : author;
+    message : message;
+    version : version;
```

Comment:
How is this new `version` field going to play with existing stores?

`V.default` is `unit` so it should be fine in that case, but as soon as we use will use something else than `unit`, the deserialisation will 💥